stylix/nixos/default.nix

```
@@ -44,7 +44,8 @@ in {
     };
   };
 
-  config = lib.optionalAttrs (options ? home-manager) (lib.mkIf (hm.autoImport && config.stylix.enable) {
-    home-manager.sharedModules = [ homeManagerModule ];
-  });
+  config.home-manager.sharedModules =
+    lib.mkIf
+    (config.stylix.enable && options ? home-manager && hm.autoImport)
```

This guard means that with `stylix.enable = false` in the NixOS config, the Home Manager modules aren't imported at all, so any Stylix options defined under Home Manager while Stylix is disabled will result in an error.

Perhaps it would be better to always import the module, then set the default of `stylix.enable` under Home Manager to follow the setting from NixOS when they are configured together, similar to what we do for other options.

---

I'll try looking into it, but I don't have much motivation or time rn.

---

> This guard means that with `stylix.enable = false` in the NixOS config, the Home Manager modules aren't imported at all, so any Stylix options defined under Home Manager while Stylix is disabled will result in an error.

Does
https://github.com/danth/stylix/commit/bb423c78bcb69c9cd9c228b131df32c7f8c63634#diff-d8c1b53a5c8dc4ca9780e130c1f21edc1f8e88589aea083b2af94fa8cb93b75eR10
already resolve this by any chance?

> Perhaps it would be better to always import the module, then set the default of `stylix.enable` under Home Manager to follow the setting from NixOS when they are configured together, similar to what we do for other options.

Would

```suggestion
    (options ? home-manager && hm.autoImport)
```

resolve this issue? Note that the original expression was replaced with an arguably simpler one. Does it accidentally change the end behaviour?

---

See https://github.com/danth/stylix/pull/244/commits/c6534b55ea718cb431e9a3afed90292971a8f5ae.

---

LGTM